File: algorithms/Arrays/Easy/Palindrome/VSDPalindrome.cpp
// Copyrights to venkys.io
// For more programs visit venkys.io 
// CPP program for Palindrome


#include <iostream>
using namespace std;

bool checkPalindrom(string str){
    int len=str.size();
    int i=0,j=len-1;
    while(i<j){
        if(str[i]!=str[j])return false;
        i++;j--;
    }
    return true;
}


int main() {
    string str="level";
    if(checkPalindrom(str))cout<<str<<" is a palindrom";
    else cout<<str<<" is not a palindrome";
    return 0;
}

File: algorithms/Arrays/Easy/Palindrome/VSDPalindrome.java
// Copyrights to venkys.io
// For more programs visit venkys.io 
// Java program for Palindrom

public class Main {
	//Function to take user input

	//Function to compute reverse of given string and return boolean value by matching it against original string
	public static boolean VSDispalindrome(String s) {
		String rev="";
		for(int i=s.length()-1;i>=0;i--) {
			rev+=s.charAt(i);
		}
		if (rev.equals(s)) return true;
		else return false;
		
	}
public static void main(String args[]) {
	String s="level";
	boolean b=VSDispalindrome(s);
	if(b)System.out.println(s+ "is Palindrome");
	else System.out.println("is not a Palindrome");
}
}
